### PR TITLE
applications: nrf_desktop: Add board documentation

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds documentation for board hardware interface.

### CODE REVIEW

**Comment by MarekPieta** [comment]
@greg-fer, could you review ?

File: applications/nrf_desktop/doc/board.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
+.. _board:
+
+Board module
+############
+
+The ``board`` module is used to assure proper state of GPIO pins which are connected to electrical components and not configured by other modules in the application.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
The ``board`` module is used to ensure the proper state of GPIO pins that are connected to electrical components and not configured by other modules in the application.
```
How does it ensure this? Mention this briefly here.

File: applications/nrf_desktop/doc/board.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+|                |             | ``board``    |                 |                  |
++----------------+-------------+--------------+-----------------+------------------+
+
+Configuration
+*************
+
+The module uses the Zephyr :ref:`zephyr:gpio` driver to set pin state.
+For this reason, you should set :option:`CONFIG_GPIO` option.
+
+For every configuration you must define ``port_state_def.h`` file (in board-specific directory in the application configuration folder).
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
For every configuration, you must define the ``port_state_def.h`` file in the board-specific directory in the application configuration folder.
```

File: applications/nrf_desktop/doc/board.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+Mentioned file defines port states by the following arrays:
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
The ``port_state_def.h`` file defines the port states by the following arrays:
```
What do you mean with `by the following arrays`? Clarify in the text.

File: applications/nrf_desktop/doc/board.rst [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+* ``port_state_on`` - state set on system start and wakeup.
+* ``port_state_off`` - state set when system enters low-power mode.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
* ``port_state_off`` - state set when the system enters the low-power mode.
```

File: applications/nrf_desktop/doc/board.rst [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+*************
+
+The module uses the Zephyr :ref:`zephyr:gpio` driver to set pin state.
+For this reason, you should set :option:`CONFIG_GPIO` option.
+
+For every configuration you must define ``port_state_def.h`` file (in board-specific directory in the application configuration folder).
+
+Mentioned file defines port states by the following arrays:
+
+* ``port_state_on`` - state set on system start and wakeup.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
* ``port_state_on`` - state set on the system start and wakeup.
```

File: applications/nrf_desktop/doc/board.rst [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+* ``port_state_off`` - state set when system enters low-power mode.
+
+Every :c:type:`struct port_state` refers to a single GPIO port and contains information about:
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
Every :c:type:`struct port_state` refers to a single GPIO port and contains the following information:
```

File: applications/nrf_desktop/doc/board.rst [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+* :cpp:member:`name` - the GPIO device name (defined by symbol in :ref:`zephyr:device-tree`).
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
* :cpp:member:`name` - GPIO device name (defined by a symbol in :ref:`zephyr:device-tree`).
```
What kind of symbol? Provide an example.

File: applications/nrf_desktop/doc/board.rst [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+* :cpp:member:`ps` - pointer to the array of :c:type:`struct pin_state`.
+* :cpp:member:`ps_count` - size of the mentioned array.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
* :cpp:member:`ps_count` - size of the `ps` array.
```

File: applications/nrf_desktop/doc/board.rst [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+Every :c:type:`struct pin state` defines state of a single GPIO pin:
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
Every :c:type:`struct pin state` defines the state of a single GPIO pin:
```

File: applications/nrf_desktop/doc/board.rst [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+* :cpp:member:`pin` - pin number.
+* :cpp:member:`val` - value which is set to pin.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
* :cpp:member:`val` - value set for the pin.
```

File: applications/nrf_desktop/doc/board.rst (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/board.rst]: This is assured by setting state of GPIO pins, I will add this information here.

File: applications/nrf_desktop/doc/board.rst (hunk 3 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/board.rst]: ok

File: applications/nrf_desktop/doc/board.rst (hunk 7 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/board.rst]: ok

File: applications/nrf_desktop/doc/board.rst [hunk 11]
```diff
@@ -0,0 +1,40 @@
+.. _board:
+
+Board module
+############
+
+The ``board`` module is used to ensure the proper state of GPIO pins that are connected to electrical components and not configured by other modules in the application.
```
Comment (pdunaj) [inline applications/nrf_desktop/doc/board.rst]: `to ensure the proper state of GPIO pins that are not configured by other modules in the application`

Comment (MarekPieta) [reply applications/nrf_desktop/doc/board.rst]: I just wanted to mention that pins that are not connected to anything may be left unset. Board module only has to configure pins that are in use (e.g. to reduce power consumption). Still, if you want I may remove this information

File: applications/nrf_desktop/doc/board.rst [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+For every configuration, you must define the ``port_state_def.h`` file in the board-specific directory in the application configuration folder.
+
+The ``port_state_def.h`` file defines the states set to the GPIO ports by the following arrays:
+
+* ``port_state_on`` - state set on the system start and wakeup.
+* ``port_state_off`` - state set when the system enters the low-power mode.
+
+Every :c:type:`struct port_state` refers to a single GPIO port and contains the following information:
+
+* :cpp:member:`name` - GPIO device name (defined by a symbol in :ref:`zephyr:device-tree` - e.g. :c:macro:`DT_GPIO_P0_DEV_NAME`).
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
* :cpp:member:`name` - GPIO device name (defined by a symbol in :ref:`zephyr:device-tree`, for example :c:macro:`DT_GPIO_P0_DEV_NAME`).
```

**Comment by MarekPieta** [comment]
@greg-fer, I made requested changes, could you rereview?

File: applications/nrf_desktop/doc/board.rst [hunk 13]
```diff
@@ -0,0 +1,40 @@
+.. _board:
+
+Board module
+############
+
+The ``board`` module is used to ensure the proper state of GPIO pins that are not configured by other modules in the application.
+This is done by setting the state of the mentioned pins on the system start, wakeup and when the system enters the low-power mode.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
Ensuring the proper state is done by setting the state of the mentioned pins on the system start and wakeup, and when the system enters the low-power mode.
```
`This` is actually not clear here...

File: applications/nrf_desktop/doc/board.rst [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
++----------------+-------------+--------------+-----------------+------------------+
+| Source Module  | Input Event | This Module  | Output Event    | Sink Module      |
++================+=============+==============+=================+==================+
+|                |             | ``board``    |                 |                  |
++----------------+-------------+--------------+-----------------+------------------+
+
+Configuration
+*************
+
+The module uses the Zephyr :ref:`zephyr:gpio` driver to set pin state.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/board.rst]:
```suggestion
The module uses the Zephyr :ref:`zephyr:gpio` driver to set the pin state.
```

File: applications/nrf_desktop/doc/board.rst (hunk 13 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/board.rst]: ok

**Comment by MarekPieta** [comment]
@pdunaj, could you take a look?

**Pull request merged**